src/mint/process.rs

```diff
@@ -89,6 +90,19 @@ pub fn process_mint(args: MintArgs) -> Result<()> {
         CANDY_EMOJI
     );
 
+    let receiver_pubkey = match args.receiver {
+        Some(receiver_id) => match Pubkey::from_str(&receiver_id) {
```

Comment:
You can make this a bit more concise by using `map_err` to convert the error from the from_str parsing error to your nice user-friendly error and then bubbling up the result with `?`:

```rust
    let receiver_pubkey = match args.receiver {
        Some(receiver_id) => Pubkey::from_str(&receiver_id)
            .map_err(|_| anyhow!("Failed to parse receiver pubkey: {}", receiver_id))?,
        None => candy_machine_state.wallet,
    };
 ```

Reply:
replaced it with your snippet, I haven't learn much Rust, just wrote this by reading the rest of the code

Reply:
All good, just wanted to give you a little tip and keep the code a little more maintainable.